driver-reactive-streams/src/main/com/mongodb/reactivestreams/client/internal/BatchCursorPublisher.java

```diff
@@ -126,7 +126,8 @@ public void subscribe(final Subscriber<? super T> subscriber) {
     }
 
     Mono<BatchCursor<T>> batchCursor(final Supplier<AsyncReadOperation<AsyncBatchCursor<T>>> supplier) {
-        return mongoOperationPublisher.createReadOperationMono(supplier, clientSession).map(BatchCursor::new);
+        // TODO: not sure this is kosher, to layer a Mono on to of a Mono with an intervening custom Publisher
+        return Mono.from(mongoOperationPublisher.createReadOperationMono(supplier, clientSession)).map(BatchCursor::new);
```

Comment:
this seems actually pretty standard (look at how eg. `Mono.map` is implemented). do you actually need the `Mono.from`? The `mongOperationPublisher#createReadOperationMono` seems to indicate a `Mono` is created... That's the only "risk": introducing an non-Reactor reactive type in the middle of the chain would prevent upstream from seeing `Context` from downstream.

Reply:
I may be misunderstanding, but it seems like the chain is going to broken no matter what, since the public driver API returns Publisher instances, not Mono or Flux instances, and applications can't assume anything about the actual implementation class of of those Publisher instances.

Reply:
indeed. Isn't the `createReadOperationMono` method controlled by you and thus guaranteed to be a `Mono`? I'm not so sure where the layering you refer to in the comment lies though, after re-reading that piece of code...

in any case, `Mono.from(x)` doesn't hurt much, since if x is a `Mono` it will be returned as is.

Reply:
@simonbasle  ok to resolve this thread?

Reply:
yes